### PR TITLE
Implemented redirections

### PR DESCRIPTION
**Pull request by gde-alme**
Might not be the best approach although it seems to work fine

### CODE REVIEW

**Comment by gde-alme** [comment]
This PR has the commits from "Issue#52 PR"

File: include/HttpRequest.hpp [hunk 1]
```diff
@@ -22,7 +22,8 @@ class HttpRequest : public HttpMessage
     // Getters & Setters
     const std::string& target() const;
     std::string& target();
-
+    void    set_target( const std::string& target );
```
Comment (PukieDiederik) [inline include/HttpRequest.hpp:25]: This function is not necessary, the target can be set by calling `target() = "new/target"`.

File: src/Server.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ Server& Server::operator=(const Server& copy)
 
 // BEGIN: Class Functions
 // Will take a request and handle it, which includes calling CGI
-HttpResponse    Server::handleRequest(const HttpRequest& req)
+HttpResponse    Server::handleRequest( HttpRequest& req)
```
Comment (PukieDiederik) [inline src/Server.cpp:47]: The request is NEVER allowed to be changed by the server

File: src/Server.cpp [hunk 3]
```diff
@@ -53,6 +53,14 @@ HttpResponse    Server::handleRequest(const HttpRequest& req)
     // Check if a valid route has been found
     if (!route) return response_error( req, res, _cfg, route, 404);
 
+    if ( route->is_redirect ) {
+        req.set_target( route->redirect_to );
+        res = Server::handleRequest( req );
```
Comment (PukieDiederik) [inline src/Server.cpp:58]: Why do we set the target and run handleRequest again? Why do we not just have the last 2 lines? The target redirect might not be for this server.

**Review by PukieDiederik — CHANGES_REQUESTED** [review]
This seems like a weird way to handle redirections. Should it not just be handled by `src/Server.cpp:59-60`?

**Review by edgareu1 — DISMISSED** [review]

**Review by edgareu1 — APPROVED** [review]

**Pull request merged**